pkg/sectigo/sectigo.go

```diff
@@ -40,16 +42,16 @@ type Sectigo struct {
 // $SECTIGO_USERNAME and $SECTIGO_PASSWORD respectively; alternatively if not given and
 // not stored in the environment, as long as valid access credentials are cached the
 // credentials will be loaded.
-func New(username, password, profile string) (client *Sectigo, err error) {
+func New(conf config.SectigoConfig) (client *Sectigo, err error) {
```

Comment:
I think that we'll have to do checking if we're in testing mode at this point and startup the Mock server if we are. There may be some issues with recursive imports though, so we'll have to figure that out. 

My suggestion is to create a module level variable `mockSectigoServer *mock.Server` and on `New` if `conf.Testing` then we create the new `mockSectigoServer`. We should create module level functions `GetMock`, `ResetMock`, and `CloseMock` that allow test code to manipulate the module level variable, checking if it is `nil` and handling it appropriately. 

I'm hoping that the only thing we'll have to change to prevent recursive imports, is to have the sectigo code set it's baseurl rather than having the mock server do it. 

Let me know if that makes sense.

Reply:
Makes sense but unfortunately resolving the recursive import looks like it's a bit more complicated. In the mock package we need access to the endpoints and the Request/Reply structs, which currently exist in the sectigo package.

Reply:
Yeh, things are super gnarly aren't they. My first instinct is to move the mock back into the `sectigo` package and have no external packages call it, but we need them to be able to modify the handlers on demand for specific tests. 

Perhaps we can chat about this tomorrow morning and see if we can figure out a solution? We might just have to move the mock back into the test code.